---
title: "diversity_extras"
output: html_document
date: "2023-08-10"
---

# Hill numbers?

I think I need to trim zeroes in another run through.. also would be nice to be able to plot mq & mw separately

```{r}
#install.packages("iNEXT")

library(iNEXT)
library(ggplot2)
#library(ggpubr)
library(phyloseq)

setwd("~/nicolagk@hawaii.edu - Google Drive/My Drive/Mosquito_business/Mosquito_microbes_git/03.microbiome_analysis/03a.diversity")

ps.all <- readRDS("../../01.data_expdesign/ps.all.clean.100.less.rds")
ps.all

##mosquito subset
ps.mq <- subset_samples(ps.all,type=="A.albopictus")
#ps.mq.no0 <- prune_taxa(taxa_sums(ps.mq)>0,ps.mq)

#ps.mq.no0

ps.mq.ol <- subset_samples(ps.mq,infusion=="OL")
ps.mq.sg <- subset_samples(ps.mq,infusion=="SG")
ps.mq.sw <- subset_samples(ps.mq,infusion=="SW")

otu.mq.ol <- t(data.frame(ps.mq.ol@otu_table))
otu.mq.sg <- t(data.frame(ps.mq.sg@otu_table))
otu.mq.sw <- t(data.frame(ps.mq.sw@otu_table))

sum.mq.ol <- c(rowSums(otu.mq.ol))
sum.mq.sg <- c(rowSums(otu.mq.sg))
sum.mq.sw <- c(rowSums(otu.mq.sw))

##water subset
ps.mw <- subset_samples(ps.all,type=="Microbial Water")
#ps.mw.no0 <- prune_taxa(taxa_sums(ps.mw)>0,ps.mw)

#ps.mw.no0

ps.mw.ol <- subset_samples(ps.mw,infusion=="OL")
ps.mw.sg <- subset_samples(ps.mw,infusion=="SG")
ps.mw.sw <- subset_samples(ps.mw,infusion=="SW")

otu.mw.ol <- t(data.frame(ps.mw.ol@otu_table))
otu.mw.sg <- t(data.frame(ps.mw.sg@otu_table))
otu.mw.sw <- t(data.frame(ps.mw.sw@otu_table))

sum.mw.ol <- c(rowSums(otu.mw.ol))
sum.mw.sg <- c(rowSums(otu.mw.sg))
sum.mw.sw <- c(rowSums(otu.mw.sw))

sum.all <- cbind(sum.mq.ol,sum.mq.sg,sum.mq.sw,sum.mw.ol,sum.mw.sg,sum.mw.sw)

#saveRDS(sum.all,"sum.all.RDS")

##taking forever - doing on cluster
#mq.inext <- iNEXT(sum.all,datatype="abundance")
```

```{r}
setwd("~/nicolagk@hawaii.edu - Google Drive/My Drive/Mosquito_business/Mosquito_microbes_git/03.microbiome_analysis/03a.diversity")

all.inext <- readRDS("inext.res.all.RDS")

ggiNEXT(all.inext)
#gg.2 <- ggiNEXT(all.inext,type=2)
```

# Phylogenetic tree

Tutorial from dada2 author [here](https://f1000research.com/articles/5-1492/v2)

Trying to get a clean fasta file - on Terminal
```
cut -d"    " -f1 sequences_100.fasta > cleaned_sequences.fasta
##note: tab doesn't copy and paste well. To insert new tab: press control and v, then press Tab

awk '!/^>/ { printf "%s", $0; n = "\n" }
/^>/ { print n $0; n = "" }
END { printf "%s", n }
' cleaned_sequences.fasta > cleaned_sequences2.fasta
##chatGPT summary of what this code should do:
##In other words, the code removes the newline character at the end of each line that does not start with ">", and adds a newline character before each line that starts with ">". This is a common operation in bioinformatics to format FASTA files. The resulting file cleaned_sequences2.fasta will have a different line wrapping format compared to the original file.

##removing hyphens and periods:
sed 's/\-//g' cleaned_sequences2.fasta > cleaned_sequences3.fasta
sed 's/\.//g' cleaned_sequences3.fasta > cleaned_sequences_done.fasta

```

```{r packages phylo d}
#install.packages('devtools')
#library(devtools)
#devtools::install_github('twbattaglia/btools')
library(btools)

ps.all <- readRDS("../../01.data_expdesign/ps.all.clean.100.less.rds")
ps.all
# ps.mq.mw <- subset_samples(ps.all,type=="A.albopictus"|type=="Microbial Water")
# ps.mq.mw.no0 <- prune_taxa(taxa_sums(ps.mq.mw)>0,ps.mq.mw)
# 
# ps.mq.mw.no0
```

Actual analysis part: 

(I'm not running the following chunk every time, because only need to generate the files once)

```{r phylo d, eval=FALSE}
library("dada2")

seqs <- getSequences("cleaned_sequences_done.fasta")
names(seqs) <- seqs # This propagates to the tip labels of the tree
#saveRDS(seqs,file="phylo.seqs.all.rds")
```

Doing this next part in the cluster because it takes forever

Script phylo.R:

```
library(dada2)
library(phangorn) 
library(DECIPHER) 

seqs <- readRDS("./phylo.seqs.all.rds")
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA)
phang.align <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phang.align)
treeNJ <- NJ(dm) # Note, tip order != sequence order
fit = pml(treeNJ, data=phang.align)

## negative edges length changed to 0!
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
                     rearrangement = "stochastic", control = pml.control(trace = 0))
detach("package:phangorn", unload=TRUE)
saveRDS(fitGTR, file="./phylo.fit.all.rds")
```

```
##job file looks like this:
#$ -V # inherit the submission environment
#$ -cwd # start job in submission directory
#$ -N phylo.sh # job name, anything you want
#$ -l h_rt=24:00:00
#$ -M thenicolakriefall@gmail.com
#$ -m be

module load R
Rscript phylo.R
##exit

##on cluster:
qsub phylo.sh
##saved output as: phylo.fit.all.rds
```

Back in R

Note: I accidentally kept switching from OTU/ASV names, the pipeline did ASV as an output (i.e. 100% clustering) but they were named Otu0001 etc.

```{r back in R}
#install.packages('devtools')
#BiocManager::install("DESeq2")
#BiocManager::install("genefilter")
#devtools::install_github('twbattaglia/btools')
library(btools)

setwd("~/nicolagk@hawaii.edu - Google Drive/My Drive/Mosquito_business/Mosquito_microbes_git/03.microbiome_analysis/03a.diversity")

##phylo tree object labeled by entire ASV sequence
fitGTR <- readRDS("phylo.fit.all.rds")

##"key" that matches ASV sequence to ASV names
taxa.seqs <- read.csv("taxa.seqs.csv",header=F)
##naming each row by ASV name
row.names(taxa.seqs) <- taxa.seqs$V1

##to check for effects of rarefying:
#ps.rare <- ps.clean 
##check ultra-trimmed dataset for the mixed models:
#ps.clean <- readRDS("../03c.mixed_models/ps.trim.mq.rds")
ps.clean <- readRDS("../03c.mixed_models/ps.trim.mw.rds")

##taxa table in current phyloseq object, has no sequence info
##ps.clean right now is the trimmed dataset [130 taxa, 416 samples]
taxa.orig <- data.frame(ps.clean@tax_table)
##adding sequence info from the "key" above
taxa <- merge(taxa.orig,taxa.seqs,by=0)

##removing unnecessary columns
taxa2 <- taxa %>%
  dplyr::select(-Row.names,-V1)
##adding row names by OTU name
row.names(taxa2) <- taxa2$OTU

##making sure the OTU names between the counts table & the taxa tables line up before naming them by the entire sequence
seqtab <- data.frame(ps.clean@otu_table)
#taxa$sqs <- row.names(taxa) 
row.names(taxa2) == colnames(seqtab)

##naming the rows by the entire sequence
row.names(taxa2) <- taxa2$V2
colnames(seqtab) <- taxa2$V2
##check they line up still
row.names(taxa2) == colnames(seqtab)
##as matrix for phyloseq's preferences
taxa.m <- as.matrix(taxa2)

samdf <- data.frame(ps.clean@sam_data)

ps.tree <- phyloseq(otu_table(seqtab, taxa_are_rows = FALSE),
                         sample_data(samdf),
                         tax_table(taxa.m),
                         phy_tree(fitGTR$tree))
ps.tree

#saveRDS(ps.tree,file="ps.clean.trim.tree.rds")
#saveRDS(ps.tree,file="ps.ultratrim.tree.mq.rds")
#saveRDS(ps.tree,file="ps.ultratrim.tree.mw.rds")

##calculating phylo diversity index
pd.div <- estimate_pd(ps.tree)
##combining with the original diversity data frame
row.names(df.div) <- df.div$sample_name
df.div.pd <- merge(df.div,pd.div,by=0)

## saving diversity data frame ##
#save & read back in as needed
#write.csv(df.div.pd,file="div.trim.phylo.csv") #saving
#df.div <- read.csv("mr16s_diversity_rev.rare_6.2k.csv",row.names=1,header=TRUE) #reading back in
```

## Plotting phylo diversity

### Mesocosm water
 
```{r}
df.div.pd.mw <- subset(df.div.pd,type=="Microbial Water")

##some variable cleaning
df.div.pd.mw$day
df.div.pd.mw$day <- sub("4","Day 4",df.div.pd.mw$day)
df.div.pd.mw$day <- sub("12","Day 12",df.div.pd.mw$day)
df.div.pd.mw$day <- sub("20","Day 20",df.div.pd.mw$day)
# df.div.pd.mw$day <- str_replace("0","Infus.",df.div.pd.mw$day)
# df.div.pd.mw$day
# str(df.div.pd.mw)
df.div.pd.mw$day <- factor(df.div.pd.mw$day,levels=c("Day 4","Day 12","Day 20"))

df.div.pd.mw$inf.temp <- paste0(df.div.pd.mw$infusion,df.div.pd.mw$temperature)

df.div.pd.mw$mesocosm <- as.factor(df.div.pd.mw$mesocosm)
df.div.pd.mw$infusion <- as.factor(df.div.pd.mw$infusion)
str(df.div.pd.mw)

df.div.pd.mw.se <- summarySE(df.div.pd.mw,measurevar="PD",groupvars=c("inf.temp","infusion","day","temperature"))

##dots & error bars
ggplot(df.div.pd.mw,aes(x=infusion,y=PD,fill=inf.temp,shape=inf.temp))+
  geom_jitter(alpha=0.5,position=position_jitterdodge(jitter.width=0.2),color="gray")+
  geom_errorbar(data=df.div.pd.mw.se,aes(ymax=PD+se,ymin=PD-se),width=0.2,color="black",position=position_dodge(width=0.6))+
  geom_point(data=df.div.pd.mw.se,size=2.5,position=position_dodge(width=0.6))+
  #geom_boxplot(outlier.shape=NA,alpha=0.5)+
  facet_wrap(~day)+
  theme_cowplot()+
  ylab("")+
  xlab("")+
  scale_fill_manual(values=c("#B5D69F","#7DBA54","#C488A2","#992559","#B6F6EE","#60eede"),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","SW_cool","SW_warm"))+
  scale_shape_manual(values=c(22,24,22,24,22,24),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","SW_cool","SW_warm"))+ 
  ggtitle("Mesocosm water")
#gg.mw.rich
```

### Mosquitoes

```{r}
df.div.pd.mq <- subset(df.div.pd,type=="A.albopictus")

df.div.pd.mq$inf.temp <- paste0(df.div.pd.mq$infusion,df.div.pd.mq$temperature)

df.div.pd.mq$sex <- sub("F","Female",df.div.pd.mq$sex,ignore.case=F)
df.div.pd.mq$sex <- sub("M","Male",df.div.pd.mq$sex,ignore.case=F)

df.div.pd.mq$inf.sex <- paste0(df.div.pd.mq$infusion,"_",df.div.pd.mq$sex)

df.div.pd.mq.se <- summarySE(df.div.pd.mq,measurevar="PD",groupvars=c("infusion","temperature","sex","inf.temp"))

##dots & error bars
ggplot(df.div.pd.mq,aes(x=infusion,y=PD,fill=inf.temp,shape=inf.temp))+
  geom_jitter(alpha=0.5,position=position_jitterdodge(jitter.width=0.2),color="gray")+
  geom_errorbar(data=df.div.pd.mq.se,aes(ymax=PD+se,ymin=PD-se),width=0.2,color="black",position=position_dodge(width=0.6))+
  geom_point(data=df.div.pd.mq.se,size=2.5,position=position_dodge(width=0.6))+
  #geom_boxplot(outlier.shape=NA,alpha=0.5)+
  facet_wrap(~sex)+
  theme_cowplot()+
  xlab("")+
  ylab("Faith's D")+
  #theme(legend.position="none")+
  scale_fill_manual(values=c("#B5D69F","#7DBA54","#C488A2","#992559","#B6F6EE","#60eede"),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","SW_cool","SW_warm"))+
  scale_shape_manual(values=c(22,24,22,24,22,24),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","SW_cool","SW_warm"))+
  #ylim(1,5.2)+
  ggtitle(" ")
```

# Archive

## All the variables

### Water richness

```{r}
df.div.mw.se <- summarySE(df.div.mw,measurevar="Observed",groupvars=c("inf.temp","infusion","day","temperature"))

##dots & error bars
gg.mw.rich <- ggplot(df.div.mw,aes(x=infusion,y=Observed,fill=inf.temp,shape=inf.temp))+
  geom_jitter(alpha=0.5,position=position_jitterdodge(jitter.width=0.2),color="gray")+
  geom_errorbar(data=df.div.mw.se,aes(ymax=Observed+se,ymin=Observed-se),width=0.2,color="black",position=position_dodge(width=0.6))+
  geom_point(data=df.div.mw.se,size=2.5,position=position_dodge(width=0.6))+
  #geom_boxplot(outlier.shape=NA,alpha=0.5)+
  facet_wrap(~day)+
  theme_cowplot()+
  ylab("")+
  xlab("")+
  scale_x_discrete(labels=c("OL","SG","PW"))+
  scale_fill_manual(values=c("#B5D69F","#7DBA54","#C488A2","#992559","#B6F6EE","#60eede"),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  scale_shape_manual(values=c(22,24,22,24,22,24),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+ 
  ggtitle("Mesocosm water")
gg.mw.rich

```

### Water, Simpson's

```{r}
df.div.mw.se <- summarySE(df.div.mw,measurevar="InvSimpson",groupvars=c("infusion","inf.temp","day","temperature"))

##dots & error bars
gg.mw.simp <- ggplot(df.div.mw,aes(x=infusion,y=InvSimpson,fill=inf.temp,shape=temperature))+
  geom_jitter(alpha=0.5,position=position_jitterdodge(jitter.width=0.2),color="gray")+
  geom_errorbar(data=df.div.mw.se,aes(ymax=InvSimpson+se,ymin=InvSimpson-se),width=0.2,color="black",position=position_dodge(width=0.6))+
  geom_point(data=df.div.mw.se,size=2.5,position=position_dodge(width=0.6))+
  #geom_boxplot(outlier.shape=NA,alpha=0.5)+
  facet_wrap(~day)+
  theme_cowplot()+
  ylab("")+
  xlab("")+
  scale_x_discrete(labels=c("OL","SG","PW"))+
  scale_fill_manual(values=c("#B5D69F","#7DBA54","#C488A2","#992559","#B6F6EE","#60eede"),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  scale_shape_manual(values=c(22,24,22,24,22,24),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  #ylim(1,5.2)+
  ggtitle(" ")
gg.mw.simp

#ggsave("gg.simp.water.pdf",width=5.5,height=3)
```

## Mosquito subset{.tabset}

```{r}
df.div.mq <- subset(df.div,type=="A.albopictus")

df.div.mq$inf.temp <- paste0(df.div.mq$infusion,df.div.mq$temperature)

df.div.mq$sex <- sub("F","Female",df.div.mq$sex,ignore.case=F)
df.div.mq$sex <- sub("M","Male",df.div.mq$sex,ignore.case=F)

df.div.mq$inf.sex <- paste0(df.div.mq$infusion,"_",df.div.mq$sex)
```

### Richness plot

```{r}
df.div.mq.se <- summarySE(df.div.mq,measurevar="Observed",groupvars=c("temperature","infusion","inf.temp"))

##dots & error bars
gg.mq.rich <- ggplot(df.div.mq,aes(x=infusion,y=Observed,fill=inf.temp,shape=inf.temp))+
  geom_jitter(alpha=0.5,position=position_jitterdodge(jitter.width=0.2),color="gray")+
  geom_errorbar(data=df.div.mq.se,aes(ymax=Observed+se,ymin=Observed-se),width=0.2,color="black",position=position_dodge(width=0.6))+
  geom_point(data=df.div.mq.se,size=2.5,position=position_dodge(width=0.6))+
  #geom_boxplot(outlier.shape=NA,alpha=0.5)+
  facet_wrap(~sex)+
  theme_cowplot()+
  xlab("")+
  ylab("ASV richness")+
  #theme(legend.position="none")+
  scale_x_discrete(labels=c("OL","SG","PW"))+
  scale_fill_manual(values=c("#B5D69F","#7DBA54","#C488A2","#992559","#B6F6EE","#60eede"),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  scale_shape_manual(values=c(22,24,22,24,22,24),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  #ylim(3,16)+
  ggtitle("Mosquitoes")
gg.mq.rich

#ggsave("gg.rich.mosq.pdf",width=4,height=3)
```

### Mosquito, simpson's

```{r}
df.div.mq.se <- summarySE(df.div.mq,measurevar="InvSimpson",groupvars=c("infusion","temperature","sex","inf.temp"))

##dots & error bars
gg.mq.simp <- ggplot(df.div.mq,aes(x=infusion,y=InvSimpson,fill=inf.temp,shape=inf.temp))+
  geom_jitter(alpha=0.5,position=position_jitterdodge(jitter.width=0.2),color="gray")+
  geom_errorbar(data=df.div.mq.se,aes(ymax=InvSimpson+se,ymin=InvSimpson-se),width=0.2,color="black",position=position_dodge(width=0.6))+
  geom_point(data=df.div.mq.se,size=2.5,position=position_dodge(width=0.6))+
  #geom_boxplot(outlier.shape=NA,alpha=0.5)+
  facet_wrap(~sex)+
  theme_cowplot()+
  xlab("")+
  ylab("Simpson's (inv.)")+
  scale_x_discrete(labels=c("OL","SG","PW"))+
  scale_fill_manual(values=c("#B5D69F","#7DBA54","#C488A2","#992559","#B6F6EE","#60eede"),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  scale_shape_manual(values=c(22,24,22,24,22,24),name="Infusion, temp.",labels=c("OL_cool","OL_warm","SG_cool","SG_warm","PW_cool","PW_warm"))+
  #ylim(1,5.2)+
  ggtitle(" ")
gg.mq.simp
# 
# ##dispersal is supposed to be significant but I don't really see it
# ggplot(df.div.mq,aes(x=sex,y=InvSimpson,color=dispersal))+
#   geom_boxplot()#+
#   facet_wrap(~sex)
```

## Show library size alongside

Helpful link: https://r-graph-gallery.com/line-chart-dual-Y-axis-ggplot2.html

```{r}
df.div.mw.lib <- summarySE(df.div.mw,measurevar="lib_size_clean",groupvars=c("day","infusion"))
df.div.mw.obs <- summarySE(df.div.mw,measurevar="Observed",groupvars=c("day","infusion"))

ggplot(df.div.mw, aes(x=infusion,fill=infusion))+
  geom_errorbar(data=df.div.mw.obs,aes(ymin=Observed-se,ymax=Observed+se),width=0.2,position=position_nudge(x=-0.2))+
  geom_point(data=df.div.mw.obs,aes(y=Observed),position=position_nudge(x=-0.2),pch=21) + # Divide by 10 to get the same range than the temperature+
  geom_errorbar(data=df.div.mw.lib,aes(ymin=(lib_size_clean-se)/1240,ymax=(lib_size_clean+se)/1240,color=infusion),width=0.2,color="darkgray")+
  geom_point(data=df.div.mw.lib,aes(y=lib_size_clean/1240,fill=infusion),color="darkgray",pch=21)+
  facet_wrap(~day)+
  scale_y_continuous(
    # Features of the first axis
    name = "ASV richness",
    # Add a second axis and specify its features
    sec.axis = sec_axis(~.*1240, name="Library size")
  )+
  theme_cowplot()+
  theme(axis.title.y.right = element_text(color="darkgray"),
        legend.position="none",
        axis.text.y.right=element_text(color="darkgray"))+
  scale_fill_manual(values=c("#7DBA54","#992559","#60eede"))+
  xlab("Infusion")
#ggsave("gg.rich.libsize.water.pdf",width=5.5,height=3)
```
