---
title: "phylo_clades"
output: html_document
date: "2023-08-02"
editor_options: 
  chunk_output_type: console
---

# Setup

```{r}
library("ape")
library("castor")
library("ggtree")
library("ggplot2")
library("dplyr")
library("glmmTMB")
library("ggpubr")

setwd("~/nicolagk@hawaii.edu - Google Drive/My Drive/Mosquito_business/mosquito_microbes/04.microbiome_analysis/04c.mixed_models/phylo_clades")
```

## Data

```{r}
ps.mq <- readRDS("../ps.trim.mq.rds")
ps.mq

##ran this next part during initial processing
##which 29 OTUs do I want
tax.mq <- data.frame(ps.mq@tax_table)

##saving the otus I want as a .txt file to extract from the fasta file
#otus.mq <- c(row.names(tax.mq))

#write.table(otus.mq, file="otus.mosquitoes.txt", append = FALSE, sep = "/n", row.names = FALSE, col.names = FALSE,quote=FALSE)
```

I have a fasta file generated by C-MAIKI pipeline that has all 2000+ ASVs, so subsetting that file for the 25 that I want. I cleaned it up in the .Rmd file for stock culture taxonomy & copied it here

```
awk -F'>' 'NR==FNR{ids[$0]; next} NF>1{f=($2 in ids)} f' otus.mosquitoes.txt sequences_100.4_copy.fasta >> otus.mosquitoes.fasta
```

Then submitted the fasta file to phylogeny.fr to get the newick tree 

# GLMM by clade

```{r}
tree <- read.tree("otus_mosquitoes.nwk.txt")
tree

##add clade into to mixed model dataset
mosqlong <- readRDS("../mosq.long.rds")

unique(tax.mq$Genus)

##add grouping info to long-form dataset
##we based 'clade' on phylo tree S2
tax.mq2 <- tax.mq %>% 
  mutate(clade=case_when(
    Genus=="Wolbachia" ~ "Wolb.",
    
    Genus=="Asaia" ~ "Alph.1",
    Genus=="Bosea" ~ "Alph.1",
    Genus=="Bradyrhizobium" ~ "Alph.1",
    Genus=="Brevundimonas" ~ "Alph.1",
    
    Genus=="Chryseobacterium" ~ "Bact.1",
    Genus=="Corynebacterium" ~ "Bact.1",
    
    Genus=="Carnobacterium" ~ "Baci.1",
    Genus=="Staphylococcus" ~ "Baci.1",
    Genus=="Tetragenococcus" ~ "Baci.1",
    
    Genus=="Pseudomonas" ~ "Gamm.1",
    Genus=="Stenotrophomonas" ~ "Gamm.1",
    
    Genus=="Kosakonia" ~ "Ente.1",
    Genus=="Klebsiella" ~ "Ente.1",
    Genus=="Family_Enterobacteriaceae" ~ "Ente.1",
    
    Genus=="Burkholderiaceae_unclassified" ~ "Beta.1",
    Genus=="Ralstonia" ~ "Beta.1",
    Genus=="Family_Comamonadaceae" ~ "Beta.1",
    Genus=="Acidovorax" ~ "Beta.1",
    Genus=="Cupriavidus" ~ "Beta.1"
  ))

##renaming tip labels with more info
# Step 1: Find indices of matching values in df2 for the common column (ID) in df1
long.tax.matches <- match(mosqlong$variable, row.names(tax.mq2))
long.tax.matches

# Step 2: Replace values in df1 with values from df2 using the matching_indices
mosqlong$clade <- tax.mq2$clade[long.tax.matches]

str(mosqlong)
mosqlong$rowid <- as.factor(mosqlong$rowid)

##full model, binomial
mosqmod.clade <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|variable)+(1|rowid)+(1|clade)+(1|mesocosm:variable)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod.clade)

##full model, with infusion
mosqmod.clade.inf <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|variable)+(1|rowid)+(1|clade)+(1|mesocosm:variable)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|clade:infusion),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod.clade.inf)

##full model, with temp
mosqmod.clade.tem <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|variable)+(1|rowid)+(1|clade)+(1|mesocosm:variable)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|clade:temperature),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod.clade.tem)

##full model, with time
mosqmod.clade.tim <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|variable)+(1|rowid)+(1|clade)+(1|mesocosm:variable)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|clade:newday),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod.clade.tim)

##full model, with sex
mosqmod.clade.sex <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|variable)+(1|rowid)+(1|clade)+(1|mesocosm:variable)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|clade:sex),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod.clade.sex)

##full model, with dispersal
mosqmod.clade.dis <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|variable)+(1|rowid)+(1|clade)+(1|mesocosm:variable)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|clade:dispersal),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod.clade.dis)

anova(mosqmod.clade,mosqmod.clade.inf) #ns 0.97
anova(mosqmod.clade,mosqmod.clade.tem) #ns 1
anova(mosqmod.clade,mosqmod.clade.tim) #ns 0.08
anova(mosqmod.clade,mosqmod.clade.sex) #ns 0.65
anova(mosqmod.clade,mosqmod.clade.dis) #ns 1

summary(mosqmod.clade)

# ##collect the conditional modes 
# #ranef(mosqmod)
# mosqmodranef <- as.data.frame(ranef(mosqmod))
# #write.csv(mosqmodranef, "mosqmodranef.csv")
# 
# ##option to remove control = messes things up
```

# Tree with bubbles

Adding conditional modes to tree. We debated whether this should be the model including interactions with clade, but the p-values suggest that term doesn't matter either way so just did the full model without clade because that's where most of the data in the manuscript comes from

```{r}
library("ape")
library("castor")
library("ggtree")
library("ggplot2")

setwd("~/Library/CloudStorage/GoogleDrive-nicolagk@hawaii.edu/My Drive/Mosquito_business/mosquito_microbes/04.microbiome_analysis/04c.mixed_models/phylo_clades")

mosq.coef <- read.csv(file="../mosqmodranef_revising.csv")

##infusion
mosq.coef.inf <- subset(mosq.coef,grpvar=="variable:infusion")

mosq.coef.inf$infusion <- substr(mosq.coef.inf$grp,9,10)
mosq.coef.inf$otu.id <- substr(mosq.coef.inf$grp,1,7)

tree <- read.tree("otus_mosquitoes.nwk.txt")
tree
tree[["tip.label"]]

##found this stuff here: https://www.biostars.org/p/405115/
df.coef.inf <- data.frame(mosq.coef.inf[,c("otu.id","infusion","condval")])
df.coef.sg <- subset(df.coef.inf,infusion=="SG")
colnames(df.coef.sg)[3] <- "condval_sg"
df.coef.sw <- subset(df.coef.inf,infusion=="SW")
colnames(df.coef.sw)[3] <- "condval_sw"
df.coef.ol <- subset(df.coef.inf,infusion=="OL")
colnames(df.coef.ol)[3] <- "condval_ol"

df.coef.sgsw <- merge(df.coef.sg,df.coef.sw,by="otu.id")
df.coef.inf.re <- merge(df.coef.sgsw,df.coef.ol,by="otu.id")

##adding more informative taxa names
tax.mq$gen_cult <- paste0(tax.mq$Genus," ",tax.mq$Pool_name)
tax.mq$gen_cult <- sub("unclassified","unc",tax.mq$gen_cult)

match.tax.coef <- match(tax.mq$OTU, df.coef.inf.re$otu.id)
df.coef.inf.re$genus <- tax.mq$gen_cult[match.tax.coef]
df.coef.inf.re$clade <- tax.mq$clade[match.tax.coef]
#tree[["tip.label"]]
#tax.mq2$Genus
matching_indices <- match(tree[["tip.label"]], df.coef.inf.re$otu.id)
tree[["tip.label"]] <- df.coef.inf.re$genus[matching_indices]
tree[["tip.label"]]

df.coef.inf.reloc <- df.coef.inf.re %>% relocate(genus)

##looking at size range values for the size scale
range(df.coef.inf$condval)

#,branch.length="none"
tree.inf <- ggtree(tree) %<+% df.coef.inf.reloc + 
  geom_tiplab(aes(x+0.085))+ 
  geom_tippoint(aes(x=x+0.02,size=condval_ol),color="#7DBA54",alpha=0.7)+
  geom_tippoint(aes(x=x+0.045,size=condval_sg),color="#992559",alpha=0.7)+
  geom_tippoint(aes(x=x+0.07,size=condval_sw),color="#60eede",alpha=0.7)+
  scale_size(range=c(0,6),limits=c(-2,3),name="Cond. mode")+
  xlim(NA,0.9)
  #geom_text(aes(x=branch,label=clade), size=3, vjust=-.3)
tree.inf

##by temperature
mosq.coef.tem <- subset(mosq.coef,grpvar=="variable:temperature")

mosq.coef.tem$temperature <- substr(mosq.coef.tem$grp,9,10)
mosq.coef.tem$otu.id <- substr(mosq.coef.tem$grp,1,7)

##found this stuff here: https://www.biostars.org/p/405115/
df.coef.temp <- data.frame(mosq.coef.tem[,c("otu.id","temperature","condval")])
df.coef.h <- subset(df.coef.temp,temperature=="H")
colnames(df.coef.h)[3] <- "condval_h"
df.coef.c <- subset(df.coef.temp,temperature=="C")
colnames(df.coef.c)[3] <- "condval_c"

df.coef.hc <- merge(df.coef.h,df.coef.c,by="otu.id")

##adding more informative taxa names
match.tax.coef.hc <- match(tax.mq2$OTU, df.coef.hc$otu.id)
df.coef.hc$genus <- tax.mq2$gen_cult[match.tax.coef.hc]
df.coef.hc$clade <- tax.mq2$clade[match.tax.coef.hc]
#tree[["tip.label"]]
#tax.mq2$Genus

df.coef.hc.reloc <- df.coef.hc %>% relocate(genus)

##looking at size range values for the size scale
range(mosq.coef.tem$condval)

#,branch.length="none"
tree.temp <- ggtree(tree) %<+% df.coef.hc.reloc + 
  geom_tiplab(aes(x+0.06))+
  geom_tippoint(aes(x=x+0.02,size=condval_h),color="brown4",alpha=0.7)+
  geom_tippoint(aes(x=x+0.045,size=condval_c),color="darkcyan",alpha=0.7)+
  #geom_tippoint(aes(x=x+0.06,size=condval_sw),color="#60eede",alpha=0.7)+
  scale_size(range=c(0,6),name="Cond. mode")+
  xlim(NA,0.9)+
  geom_text(aes(x=branch,label=clade), size=3, vjust=-.3)
tree.temp

ggarrange(tree.inf,tree.temp,nrow=2,labels=c("(a)","(b)"))
#ggsave("phylo.clades.inf.temp.pdf",height=9,width=8)
```

# Archive

## Plotting

```{r}
foo <- function() {
  for (i in 1:2)
    axis(i, las = 1)
  box(lty = "19")
}

# plot(tree); foo()
# abline(v=0.12,col="red")

#tree.12 <- split_tree_at_height(tree, height=0.12, by_edge_count=FALSE)
```

