---
title: "Beta diversity mixed models"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
 rmarkdown::html_document:
    theme: cerulean
    toc: yes
    toc_float: yes
    highlight: haddock
    number_sections: true
editor_options: 
  chunk_output_type: console
---

# Setup

## Libs

```{r}
#install.packages("remotes")
#install.packages('TMB',source=TRUE) #didn't work on home laptop
#remotes::install_github("glmmTMB/glmmTMB/glmmTMB")
library("glmmTMB")
#install.packages("BiocManager")
#BiocManager::install("phyloseq")
library("phyloseq")
library("dplyr")
#install.packages("reshape")
library("reshape")
library("ggplot2")
#install.packages("bbmle")
library("bbmle")
#install.packages("car")
library("car")
```

## Re-read data

```{r}
setwd("~/nicolagk@hawaii.edu - Google Drive/My Drive/Mosquito_business/mosquito_microbes/04.microbiome_analysis/04c.mixed_models")

ps.clean <- readRDS("../../02.process_asvs/ps.clean.trim.less.rds")
#ps.clean <- readRDS("../../01.data_expdesign/ps.all.clean.100.less.rds")
ps.clean #161 taxa, 458 samples

##mosquito things - made below
#ps.trim.mq <- readRDS("ps.trim.mq.rds")
# ps.trim.mq
# mosqlong <- readRDS("mosq.long_old.rds")
# mosqlong.tax <- readRDS("mosq.long.tax_old.rds")
# 
# ps.trim.mw <- readRDS("ps.trim.mw.rds")
# ps.trim.mw
# mwlong <- readRDS("mw.long.rds")
# mwlong.tax <- readRDS("mw.long.tax.rds")
```

# Data formatting

## Mosquitoes

### Trim, thoroughly

Ran once then saved

```{r}
ps.mq <- subset_samples(ps.clean,type=="A.albopictus")
ps.mq

ps.trim.mq <- filter_taxa(ps.mq, function(x) sum(x > 1) > (0.035*length(x)), TRUE)
ps.trim.mq #29 taxa
0.035*195
#3.5% of 195 samples is >6.8 samples

ps.trim.mq2 <- prune_samples(sample_sums(ps.trim.mq) >= 1, ps.trim.mq)
ps.trim.mq2
#no samples missing data, not using this phyloseq object

##proportion of zeroes
otu.trim.mq <- data.frame(ps.trim.mq@otu_table)
sum(otu.trim.mq == 0) / (ncol(otu.trim.mq) * nrow(otu.trim.mq)) * 100
#72.37843%

otu.clean.mq <- data.frame(ps.mq@otu_table)
sum(otu.clean.mq == 0) / (ncol(otu.clean.mq) * nrow(otu.clean.mq)) * 100
#94.63927

##save if ready
#saveRDS(ps.trim.mq,file="ps.trim.mq.rds")
```

### More formatting - long form

```{r}
#total read count colums
otu.trim.mq$sum <- rowSums(otu.trim.mq)
#subset samples with >5 reads (get rid of zero samples)
otu.trim.mq.less <- subset(otu.trim.mq, sum>5) #none removed
#create sample name column for joining
otu.trim.mq.less$sample_name <- row.names(otu.trim.mq.less)

#bring in the metadata again
sam.mq <- data.frame(ps.trim.mq@sam_data)

#join the data sets by sample name
mq.tot1 <- plyr::join(otu.trim.mq.less,sam.mq, by="sample_name", type="left")
colnames(mq.tot1)

##getting rid of some irrelevant metadata
mq.tot <- mq.tot1 %>%
  dplyr::select(-orgname,-X16scq,-date.collected,-day,-stage,-abdomen.length.mm,-wing.length,-hindleg.length,-special,-Wolb_ct1,-Wolb_ct2,-Act_ct,-notes,-lib_size,-is.neg,-raw_miseq_reads)
colnames(mq.tot)

mosqlong <- melt(mq.tot, id.vars=c("sum", "sample_name", "mesocosm", "type", "sex", "dispersal", "temperature", "infusion", "newday"))

#add unique id for each row in the data. useful to control for overdispersion in models that assume error distributions that cannot account for it
mosqlong$rowid <- 1:nrow(mosqlong)
#make a long form dataset with taxonomic info
tax.mq <- data.frame(ps.trim.mq@tax_table)
tax.mq$variable <- tax.mq$OTU
mosqlong.tax <- plyr::join(mosqlong,tax.mq, by="variable", type="left")

##save
#saveRDS(mosqlong,file="mosq.long.rds")
#saveRDS(mosqlong.tax,file="mosq.long.tax.rds")
```

## Water

### Trim, thoroughly

Ran once then saved

```{r}
ps.mw <- subset_samples(ps.clean,type=="Microbial Water")
ps.mw

ps.trim.mw1 <- filter_taxa(ps.mw, function(x) sum(x > 1) > (0.035*length(x)), TRUE)
ps.trim.mw1
211*0.035 #>7.4 samples
#211 samples

ps.trim.mw <- prune_samples(sample_sums(ps.trim.mw1) >= 1, ps.trim.mw1)
ps.trim.mw #39 taxa, 211 samples

##proportion of zeroes after
otu.trim.mw <- data.frame(ps.trim.mw@otu_table)
sum(otu.trim.mw == 0) / (ncol(otu.trim.mw) * nrow(otu.trim.mw)) * 100
#44.61052

##before
otu.clean.mw <- data.frame(ps.mw@otu_table)
sum(otu.clean.mw == 0) / (ncol(otu.clean.mw) * nrow(otu.clean.mw)) * 100
#86.49142

##save if ready
#saveRDS(ps.trim.mw,file="ps.trim.mw.rds")
```

### More formatting - long form

```{r}
#total read count colums
otu.trim.mw$sum <- rowSums(otu.trim.mw)
#subset samples with >5 reads (get rid of zero samples)
otu.trim.mw.less <- subset(otu.trim.mw, sum>5) #none removed
#create sample name column for joining
otu.trim.mw.less$sample_name <- row.names(otu.trim.mw.less)

#bring in the metadata again
sam.mw <- data.frame(ps.trim.mw@sam_data)

#join the data sets by sample name
mw.tot1 <- plyr::join(otu.trim.mw.less,sam.mw, by="sample_name", type="left")
colnames(mw.tot1)

##getting rid of some irrelevant metadata
mw.tot <- mw.tot1 %>%
  dplyr::select(-orgname,-X16scq,-date.collected,-stage,-abdomen.length.mm,-wing.length,-hindleg.length,-special,-Wolb_ct1,-Wolb_ct2,-Act_ct,-notes,-lib_size,-is.neg,-sex,-newday,-raw_miseq_reads)
colnames(mw.tot)

mwlong <- melt(mw.tot, id.vars=c("sum", "sample_name","mesocosm","type", "day", "dispersal", "temperature", "infusion"))
mwlong$day <- as.factor(mwlong$day)

#add unique id for each row in the data. useful to control for overdispersion in models that assume error distributions that cannot account for it
mwlong$rowid <- 1:nrow(mwlong)
#make a long form dataset with taxonomic info
tax.mw <- data.frame(ps.trim.mw@tax_table)
tax.mw$variable <- tax.mw$OTU
mwlong.tax <- plyr::join(mwlong,tax.mw, by="variable", type="left")

##save
#saveRDS(mwlong,file="mw.long.rds")
#saveRDS(mwlong.tax,file="mw.long.tax.rds")
```

# Running the statz

## Mosquitoes

- variable = Otu name
- row id = every single entry
- value = the counts for the OTU
- sum = total counts per sample

```{r}
str(mosqlong)
mosqlong$newday <- as.factor(mosqlong$newday)
mosqlong$mesocosm <- as.factor(mosqlong$mesocosm)
mosqlong$sex <- as.factor(mosqlong$sex)
mosqlong$dispersal <- as.factor(mosqlong$dispersal)
mosqlong$temperature <- as.factor(mosqlong$temperature)
mosqlong$infusion <- as.factor(mosqlong$infusion)
mosqlong$rowid <- as.factor(mosqlong$rowid)
str(mosqlong)

##full model, binomial
mosqmod <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|variable),control=glmmTMBControl(profile=TRUE),family=binomial,data=mosqlong)
summary(mosqmod)

##no dispersal
mosqmod.nodis <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|variable),family=binomial,control=glmmTMBControl(profile=TRUE),data=mosqlong)

##no day
mosqmod.noday <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:dispersal)+(1|variable:temperature)+(1|variable:infusion)+(1|variable:sex)+(1|variable),family=binomial,control=glmmTMBControl(profile=TRUE),data=mosqlong)

##no temp
mosqmod.notem <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:infusion)+(1|variable:sex)+(1|variable),family=binomial,control=glmmTMBControl(profile=TRUE),data=mosqlong)

##no infusion
mosqmod.noinf <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:sex)+(1|variable),family=binomial,control=glmmTMBControl(profile=TRUE),data=mosqlong)

##no sex
mosqmod.nosex <- glmmTMB(cbind(value,sum-value)~dispersal+newday+temperature+infusion+sex+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:dispersal)+(1|variable:newday)+(1|variable:temperature)+(1|variable:infusion)+(1|variable),family=binomial,control=glmmTMBControl(profile=TRUE),data=mosqlong)

#Anova(mosqmod)

anova(mosqmod,mosqmod.nodis) ##0.3644
anova(mosqmod,mosqmod.noday) #sig #0.05
anova(mosqmod,mosqmod.notem) #0.05356 .
anova(mosqmod,mosqmod.noinf) #sig 0.001 
anova(mosqmod,mosqmod.nosex) #sig 0.001

summary(mosqmod)
# Conditional model:
#  Groups               Name        Variance Std.Dev.
#  mesocosm             (Intercept)  0.40861 0.6392  
#  mesocosm:variable    (Intercept)  1.19199 1.0918  
#  rowid                (Intercept)  8.21730 2.8666  
#  variable:dispersal   (Intercept)  0.06734 0.2595  
#  variable:newday      (Intercept)  0.37080 0.6089  
 # variable:temperature (Intercept)  0.20690 0.4549  
 # variable:infusion    (Intercept)  2.33338 1.5275  
 # variable:sex         (Intercept)  0.25946 0.5094  
 # variable             (Intercept) 19.24924 4.3874  

##collect the conditional modes 
#ranef(mosqmod)
mosqmodranef <- as.data.frame(ranef(mosqmod))
#write.csv(mosqmodranef, "mosqmodranef_revising.csv")

##option to remove control = messes things up

#saveRDS(mosqmod,file="mosq.model.rds")
```

## Mesocosm water

```{r}
str(mwlong)
mwlong$day <- as.factor(mwlong$day)
mwlong$mesocosm <- as.factor(mwlong$mesocosm)
mwlong$dispersal <- as.factor(mwlong$dispersal)
mwlong$temperature <- as.factor(mwlong$temperature)
mwlong$infusion <- as.factor(mwlong$infusion)
mwlong$rowid <- as.factor(mwlong$rowid)
str(mwlong)

##full model, binomial
mwmod <- glmmTMB(cbind(value,sum-value)~dispersal+day+temperature+infusion+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:infusion)+(1|variable:dispersal)+(1|variable:temperature)+(1|variable:day)+(1|variable), family=binomial, data=mwlong,control=glmmTMBControl(profile=TRUE))
summary(mwmod)

##no dispersal
mwmod.nodis <- glmmTMB(cbind(value,sum-value)~dispersal+day+temperature+infusion+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:infusion)+(1|variable:temperature)+(1|variable:day)+(1|variable), family=binomial, data=mwlong,control=glmmTMBControl(profile=TRUE))

##no day
mwmod.noday <- glmmTMB(cbind(value,sum-value)~dispersal+day+temperature+infusion+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:infusion)+(1|variable:dispersal)+(1|variable:temperature)+(1|variable), family=binomial, data=mwlong,control=glmmTMBControl(profile=TRUE))

##no temp
mwmod.notem <- glmmTMB(cbind(value,sum-value)~dispersal+day+temperature+infusion+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:infusion)+(1|variable:dispersal)+(1|variable:day)+(1|variable), family=binomial, data=mwlong,control=glmmTMBControl(profile=TRUE))

##no infusion
mwmod.noinf <- glmmTMB(cbind(value,sum-value)~dispersal+day+temperature+infusion+(1|mesocosm)+(1|mesocosm:variable)+(1|rowid)+(1|variable:dispersal)+(1|variable:temperature)+(1|variable:day)+(1|variable), family=binomial, data=mwlong,control=glmmTMBControl(profile=TRUE))

anova(mwmod,mwmod.nodis) #sig 0.001
anova(mwmod,mwmod.noday) #sig 0.001
anova(mwmod,mwmod.notem) #sig 0.001
anova(mwmod,mwmod.noinf) #sig 0.001

summary(mwmod)
# Conditional model:
#  Groups               Name        Variance Std.Dev.
#  mesocosm             (Intercept)  0.04206 0.2051  
#  mesocosm:variable    (Intercept)  0.21642 0.4652  
#  rowid                (Intercept)  2.04435 1.4298  
#  variable:infusion    (Intercept)  8.14185 2.8534  
#  variable:dispersal   (Intercept)  0.07331 0.2708  
#  variable:temperature (Intercept)  0.27815 0.5274  
#  variable:day         (Intercept)  2.50187 1.5817  
#  variable             (Intercept) 12.25267 3.5004  

#collect the conditional modes for the best fit binomial model
#ranef(mwmod)
mwmodranef <- as.data.frame(ranef(mwmod))
#write.csv(mwmodranef, "mwmodranef_revising.csv")

#saveRDS(mwmod,file="water.model.rds")
```


